src/Serializer/CustomContextBuilder.php

```diff
@@ -58,10 +61,47 @@ private function extractPathParameters(Request $request, array &$context): void
             ARRAY_FILTER_USE_KEY
         );
 
-        $context['path_parameters'] = array_map(
-            static fn (string $pathParameter) => StringUtil::castFromString($pathParameter),
-            $pathParameters
-        );
+        /**
+         * @var string $pathParameterName
+         * @var string $pathParameterValue
+         */
+        foreach ($pathParameters as $pathParameterName => $pathParameterValue) {
+            /** @var class-string $resourceClass */
+            $resourceClass = $context['resource_class'];
+
+            $propertyType = $this->extactPropertyTypeFromResourceClass($resourceClass, $pathParameterName);
+
+            if ($propertyType !== null) {
+                settype($pathParameterValue, $propertyType);
```

Comment:
This won't work if the propertyType is a float or double, because the schema is a json schema and will return number for float or double, so we will need to do the mapping if that's the case.

Reply:
See: https://json-schema.org/understanding-json-schema/reference/type.html and https://www.php.net/manual/en/function.settype.php

Reply:
https://github.com/arnedesmedt/api-platform-event-engine-bundle/pull/7/files#diff-fef89967f284307894d03ef6d71decc75b54dacbd7bc0c9d4647c30d5c1e9d2dR101